src/format/reveal/format-reveal.ts

```diff
@@ -475,6 +478,15 @@ function revealHtmlPostprocessor(
         }
       }
 
+      // bugfix for #6800
+      // if slides have content that was added by quarto then preserve that content
+      const slideContentFromQuarto = (slide as Element).querySelector(
+        ".quarto-auto-generated-content",
+      );
+      if (slideContentFromQuarto) {
+        slide.parentNode?.appendChild(slideContentFromQuarto);
+      }
```

Comment:
> Placing it in the previous slide (which will become the last one) would be using (slide as Element).previousElementSibling - so an adaptation to your alternative 1.

I thought about that, but then we have to make sure nothing will ever cause the previousElementSibling to not be a section element, etc.

> And also, at the end, there will be a new empty div.quarto-auto-generated-content in the browser DOM after all revealjs tweaks. Probably not an issue but it still feel weird to me to have empty node like this. 🤷‍♂️

In my view it's much better to have a uniform DOM structure than to have special cases in our code. If you remove the empty node, then every future call to `document.querySelector("div.quarto-auto-generated-content")` will have to check for null; if you don't remove it, then you know the element will exist.

Reply:
> I thought about that, but then we have to make sure nothing will ever cause the previousElementSibling to not be a section element, etc.

I was under the assumption that 
* We target `slide` element which will be `<section>`
* All slides `<section>` are under a `div.slides`
* Our markdown insertion are after body, meaning after last slide 
* So I assume previousElementSibling would be empty only if the presentation as only one slide which is hidden - not sure that would happen (and in that case no footer needed). Otherwise it would always be a `<section>` which are the expected sibbling at this level of the DOM. 

> In my view it's much better to have a uniform DOM structure than to have special cases in our code.

I understand that; It is just that we are adding  a div specifically for the purpose of wrapping some custom content that we insert through markdown body, and move later in the DOM thanks to a reveal plugin. So we add something that has no meaning regarding to the content, and it meant to be empty. I made my comment with that in mind. 

Anyhow, all good. Nothing to change then. Just that maybe alternative 2 is better to keep DOM uniform.